R/resourceControl.R

```diff
@@ -1,43 +1,95 @@
 #' Controlling the resource parameters in the tuning gadget
 #' @inheritParams abundanceControl
 resourceControl <- function(input, output, session, params, flags, ...) {
+
     observe({
+        re <- isolate(input$re)
+        p <- isolate(params())
         req(input$kappa,
             input$lambda,
             input$log_r_pp,
             input$w_pp_cutoff,
             input$n_resource)
-        p <- setResource(isolate(params()),
-                         kappa = input$kappa,
-                         lambda = input$lambda,
-                         r_pp = 10^input$log_r_pp,
-                         w_pp_cutoff = input$w_pp_cutoff,
-                         n = input$n_resource)
-        mu <- getResourceMort(p)
-        p@initial_n_pp <- p@rr_pp * p@cc_pp / (p@rr_pp + mu)
+        if(!is.null(getComponent(p, "MR")))
+        {
+            sel_re <- which(p@other_params$other$MR$resource_params$resource == re)
+            mizerMR::resource_params(p)$kappa[sel_re]  <- input$kappa
+            mizerMR::resource_params(p)$lambda[sel_re] <- input$lambda
+            mizerMR::resource_params(p)$r_pp[sel_re]   <- (10^input$log_r_pp)
+            mizerMR::resource_params(p)$w_max[sel_re]  <- input$w_pp_cutoff
+            mizerMR::resource_params(p)$n[sel_re]      <- input$n_resource
+            mizerMR::resource_params(p)$w_min[sel_re]  <- input$w_min_resource
+            mizerMR::initialNResource(p) <- mizerMR::resource_capacity(p)
```

Comment:
I see you set the resource spectrum back to its carrying capacity. That could be quite a large change away from its steady state spectrum and make it difficult to find the steady state again. I think you should set the resource spectrum to its new steady state value, as is done in the single-resource code.